packages/dev/core/src/Meshes/meshUVSpaceRenderer.ts

```diff
@@ -145,36 +195,152 @@ export class MeshUVSpaceRenderer {
      * @param angle The rotation angle around the direction of the projection
      */
     public renderTexture(texture: BaseTexture, position: Vector3, normal: Vector3, size: Vector3, angle = 0): void {
+        // Create the diffuse render target texture if it doesn't exist
+        if (!this.texture && !this._options.uvEdgeBlending && !this._decalTexture) {
+            this._updateRTT();
+        } else if (this.texture && !this._userCreatedTextureSetup && !this._options.uvEdgeBlending && !this._decalTexture) {
+            this._updateRTT();
+        } else if (this.texture && !this._userCreatedTextureSetup && this._options.uvEdgeBlending && !this._decalTexture) {
+            this._createDecalDiffuseRTT();
+        } else if (!this.texture && this._options.uvEdgeBlending && !this._decalTexture) {
+            this._createDecalDiffuseRTT();
+        }
+
+        // // Prepare the shader with the decal texture, mask texture, and necessary uniforms
+        const shader = MeshUVSpaceRenderer._GetShader(this._scene);
+        shader.setTexture("textureSampler", texture); // Decal texture
+
+        // Calculate and set the projection matrix
+        const projectionMatrix = this._createProjectionMatrix(position, normal, size, angle);
+        shader.setMatrix("projMatrix", projectionMatrix);
+
+        if (!this._options.uvEdgeBlending) {
+            if (this.texture instanceof RenderTargetTexture) {
+                this.texture.render();
+            }
+        } else {
+            if (MeshUVSpaceRenderer._IsRenderTargetTexture(this._decalTexture)) {
+                if (this._options.uvEdgeBlending) {
+                    this._decalTexture.render();
+                    this._createMaskTexture();
+                    this._createFinalTexture();
+                }
+            }
+        }
+    }
+    /**
+     * Creates a texture RTT if one doesn't exist,
+     */
+    _updateRTT() {
         if (!this.texture) {
-            this._createDiffuseRTT();
+            this.texture = this._createRenderTargetTexture(this._options.width, this._options.height);
+        } else {
+            this._userCreatedTextureSetup = true;
+        }
+        if (this.texture instanceof RenderTargetTexture) {

```

Comment:
Should it not check _userCreatedTextureSetup instead so if the user provides an RTT it does not break its setup ?